### PR TITLE
WIP: Hydropower example

### PR DESCRIPTION
**Pull request by jetuk**
If someone has a few minutes it would be good to get an opinion on this new example.

Several requests have come in for hydropower. We should have a good answer to how this is done in pywr.

### CODE REVIEW

**Comment by jetuk** [comment]
This all fixes some issues with loading storage levels from JSON.

File: pywr/nodes.py [hunk 1]
```diff
@@ -401,6 +404,11 @@ def load(cls, data, model):
             cost = 0.0
         node.cost = cost
 
+        if level is not None:
+            level = load_parameter(model, level)
+        print(level)
```
Comment (snorfalorpagus) [inline pywr/nodes.py]: Remove.

File: pywr/parameters/_parameters.pyx [hunk 2]
```diff
@@ -305,7 +305,6 @@ cdef class TablesArrayParameter(IndexParameter):
 
         self.h5store = H5Store(self.h5file, None, "r")
         node = self.h5store.file.get_node(self.where, self.node)
-
```
Comment (snorfalorpagus) [inline pywr/parameters/_parameters.pyx]: This will conflict with #490

**Comment by jetuk** [comment]
Cool. I'll update this. 

Do you think the `HydroPowerRecorder` should be in the main library?

**Comment by jetuk** [comment]
@aphurford this is the example we talked about.

**Comment by jetuk** [comment]
I've rebased this against current master and fixed the comments.

**Pull request merged**